includes/admin/class-wc-admin-dashboard.php

```diff
@@ -201,45 +201,33 @@ private function status_widget_order_rows() {
 		private function status_widget_stock_rows() {
 			global $wpdb;
 
-			// Get products using a query - this is too advanced for get_posts.
-			$stock          = absint( max( get_option( 'woocommerce_notify_low_stock_amount' ), 1 ) );
-			$nostock        = absint( max( get_option( 'woocommerce_notify_no_stock_amount' ), 0 ) );
-			$transient_name = 'wc_low_stock_count';
+			$stock   = absint( max( get_option( 'woocommerce_notify_low_stock_amount' ), 1 ) );
+			$nostock = absint( max( get_option( 'woocommerce_notify_no_stock_amount' ), 0 ) );
 
+			$transient_name   = 'wc_low_stock_count';
 			$lowinstock_count = get_transient( $transient_name );
+
 			if ( false === $lowinstock_count ) {
-				$query_from       = apply_filters(
-					'woocommerce_report_low_in_stock_query_from',
-					"FROM {$wpdb->posts} as posts
-					INNER JOIN {$wpdb->postmeta} AS postmeta ON posts.ID = postmeta.post_id
-					INNER JOIN {$wpdb->postmeta} AS postmeta2 ON posts.ID = postmeta2.post_id
-					WHERE 1=1
-					AND posts.post_type IN ( 'product', 'product_variation' )
-					AND posts.post_status = 'publish'
-					AND postmeta2.meta_key = '_manage_stock' AND postmeta2.meta_value = 'yes'
-					AND postmeta.meta_key = '_stock' AND CAST(postmeta.meta_value AS SIGNED) <= '{$stock}'
-					AND postmeta.meta_key = '_stock' AND CAST(postmeta.meta_value AS SIGNED) > '{$nostock}'"
+				$lowinstock_count = (int) $wpdb->get_var(
+					$wpdb->prepare(
+						"SELECT COUNT( product_id ) FROM {$wpdb->wc_product_meta_lookup} WHERE stock <= %d AND stock > %d",
```

Comment:
This code can run before the new table is populated and, if that happens, it will return wrong results. Should we add checks here and every other place where we query the new table to make sure the database update routine was executed and `wp_wc_product_meta_lookup` populated?

Reply:
I've added a version compare for this widget and price filter so they are not displayed until after updating.

Sorting should be left as is. Missing data does not prevent results showing and we have no fallback.